mapsforge-map/src/main/java/org/mapsforge/map/layer/renderer/DatabaseRenderer.java

```diff
@@ -67,7 +69,7 @@ public class DatabaseRenderer extends StandardRenderer {
      * @param hillsRenderConfig the hillshading setup to be used (can be null).
      */
     public DatabaseRenderer(MapDataStore mapDataStore, GraphicFactory graphicFactory, TileCache tileCache,
-                            TileBasedLabelStore labelStore, boolean renderLabels, boolean cacheLabels,
+                            LabelStore labelStore, boolean renderLabels, boolean cacheLabels,
```

Comment:
@Sublimis Why is this change?
As it is always a `TileBasedLabelStore`.

Reply:
It's not always -- even in the current master it can be `null`. `DatabaseRenderer` does not depend on this being a `TileBasedLabelStore`. Current implementation of `processLabels()` also does not *depend* on a particular `LabelStore` implementation, it only assumes there's a `LabelStore` that contains some labels, nothing more. It can even run with `labelStore == null`.

And now that `DirectRenderer` extends `DatabaseRenderer` in this PR, LabelStore implementation can also be `MapDataStoreLabelStore`, so it feels natural to generalize things. If this is really something you want to avoid we can for example create a new interface extending `LabelStore` that would define the `TileBasedLabelStore.storeMapItems()`. Otherwise too much code would get duplicated in `DirectRenderer` (there would be no point in extending `DatabaseRenderer` if it can only work with `TileBasedLabelStore`).